src/devices/efergy_e2_classic.c

```diff
@@ -60,24 +68,59 @@ static int efergy_e2_classic_callback(bitbuffer_t *bitbuffer) {
 		return 0;
 	}
 
-	const unsigned VOLTAGES[] = {110, 115, 120, 220, 230, 240, 0};
-
-	double current_adc = 256 * bytes[4] + bytes[5];
-	for (unsigned i = 0; VOLTAGES[i] != 0; ++i) {
-		double power  = (VOLTAGES[i] * current_adc * (1 << bytes[6])) / 32768;
-		fprintf(stderr, "Power consumption at %u volts: %.2f watts\n", VOLTAGES[i], power);
+	if (bytes[6] > 0x80)
```

Comment:
Though I know a fair bit about C, and when in doubt am not afraid to look at the assembler output, I am wrong on occasion or will at least not explain clearly enough :)
As for other peoples device code here it's best to assume that it is tweaked to work, but mostly with bad assumptions… C isn't the "native" language for most, and even when it is, C is designed to bite you…